ws-client/src/client.rs

```diff
@@ -167,8 +175,14 @@ impl WsClient {
 				max_concurrent_requests,
 			)
 			.await;
+			has_shutdown_sender.store(true, Ordering::Relaxed);
 		});
-		Ok(Self { to_back, request_timeout, error: Mutex::new(ErrorFromBack::Unread(err_rx)) })
+		Ok(Self { to_back, request_timeout, error: Mutex::new(ErrorFromBack::Unread(err_rx)), has_shutdown })
+	}
+
+	/// Checks if the client is connected to the target.
+	pub fn is_connected(&self) -> bool {
+		!self.has_shutdown.load(Ordering::Relaxed)
```

Comment:
I think (I'm not 100% sure) we could drop the atomic bool entirely and just check if the sender to the background task is still operational here?

```rust
!self.to_back.is_closed()
```

Reply:
Sorry that I didn't spotted this, pushed a PR to your branch with a fix.